hs_core/hydroshare/resource.py

```diff
@@ -376,7 +382,24 @@ def create_resource(
         if not metadata:
             metadata = []
 
-        add_resource_files(resource.short_id, *files)
+        if len(files) == 1 and unpack_file and zipfile.is_zipfile(files[0]):
+            # Add contents of zipfile asynchronously; wait 30 seconds to be "sure" that resource creation
+            # has finished.
+            import logging
+            logger = logging.getLogger('django')
+            tmp_dir = '/shared_temp'
+            logger.debug("Copying uploaded file from {0} to {1}".format(files[0].temporary_file_path(),
+                                                                        tmp_dir))
+            shutil.copy(files[0].temporary_file_path(), tmp_dir)
+            zfile_name = os.path.join(tmp_dir, os.path.basename(files[0].temporary_file_path()))
+            logger.debug("Retained upload as {0}".format(zfile_name))
+            add_zip_file_contents_to_resource.apply_async((resource.short_id, zfile_name),
+                                                          countdown=30)
+            resource.file_unpack_status = 'Pending'
+            resource.save()
+        else:
+            # Add resource file(s) now
+            add_resource_files(resource.short_id, *files)
```

Comment:
why is handling the zipfile async but handling a bunch of files that weren't compressed is not?